# MODELOS DE PREDICCION DE PEPTIDOS ANTIMICROBIANOS

## Librerias

In [114]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline
from sklearn.preprocessing import StandardScaler
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score
from sklearn.model_selection import train_test_split
from sklearn.linear_model import Lasso
from sklearn.feature_selection import SelectFromModel

## Cargar Datos

In [88]:
#Se cargan los datos negativos
dataN = pd.read_csv('./Datos/DatosNegativos2.csv')

In [89]:
#Se cargan los datos positivos
dataP = pd.read_csv('./Datos/DatosPositivos2.csv')

In [90]:
print("El archivo con datos Negativos tiene {} filas y {} columnas".format(dataN.shape[0],dataN.shape[1]))
print("El archivo con datos Positivos tiene {} filas y {} columnas".format(dataP.shape[0],dataP.shape[1]))

El archivo con datos Negativos tiene 5561 filas y 1761 columnas
El archivo con datos Positivos tiene 8322 filas y 1761 columnas


In [91]:
pd.options.display.max_columns = None
# Modificación de las opciones de pandas para mostrar todas las columnas del dataframe

In [92]:
# Verificación de campos
dataP.head(2)

Unnamed: 0             sequence  length  molecular_weight  charge  \
0           0  CGGYSGGWKRKRSTSYRCG      19           2109.36   4.859   
1           1     GIMISLMKKLAAHIAK      16           1725.22   3.094   

   charge_density  isoelectric_point     gravy  instability_index  \
0        0.002304          10.056702 -1.310526          73.031579   
1        0.001793          10.302063  0.887500          20.918750   

   aromaticity  aliphatic_index  boman_index  hydrophobic_ratio      A  \
0     0.157895            0.000     3.122105           0.105263   0.00   
1     0.000000          140.625    -0.685000           0.625000  18.75   

        R    N    D       C    E    Q       G     H      I     L       K  \
0  15.789  0.0  0.0  10.526  0.0  0.0  26.316  0.00   0.00   0.0  10.526   
1   0.000  0.0  0.0   0.000  0.0  0.0   6.250  6.25  18.75  12.5  18.750   

      M    F    P       S      T      W       Y    V    AA   AR   AN   AD  \
0   0.0  0.0  0.0  15.789  5.263  5.263  10.526  0.0  0.00  0.0  0.0  0.0   
1  12.5  0.0  0.0   6.250  0.000  0.000   0.000  0.0  6.67  0.0  0.0  0.0   

    AC   AE   AQ   AG    AH   AI   AL    AK   AM   AF   AP   AS   AT   AW  \
0  0.0  0.0  0.0  0.0  0.00  0.0  0.0  0.00  0.0  0.0  0.0  0.0  0.0  0.0   
1  0.0  0.0  0.0  0.0  6.67  0.0  0.0  6.67  0.0  0.0  0.0  0.0  0.0  0.0   

    AY   AV   RA   RR   RN   RD    RC   RE   RQ   RG   RH   RI   RL    RK  \
0  0.0  0.0  0.0  0.0  0.0  0.0  5.56  0.0  0.0  0.0  0.0  0.0  0.0  5.56   
1  0.0  0.0  0.0  0.0  0.0  0.0  0.00  0.0  0.0  0.0  0.0  0.0  0.0  0.00   

    RM   RF   RP    RS   RT   RW   RY   RV   NA   NR   NN   ND   NC   NE   NQ  \
0  0.0  0.0  0.0  5.56  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
1  0.0  0.0  0.0  0.00  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   

    NG   NH   NI   NL   NK   NM   NF   NP   NS   NT   NW   NY   NV   DA   DR  \
0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
1  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   

    DN   DD   DC   DE   DQ   DG   DH   DI   DL   DK   DM   DF   DP   DS   DT  \
0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
1  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   

    DW   DY   DV   CA   CR   CN   CD   CC   CE   CQ     CG   CH   CI   CL  \
0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  11.11  0.0  0.0  0.0   
1  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   0.00  0.0  0.0  0.0   

    CK   CM   CF   CP   CS   CT   CW   CY   CV   EA   ER   EN   ED   EC   EE  \
0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
1  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   

    EQ   EG   EH   EI   EL   EK   EM   EF   EP   ES   ET   EW   EY   EV   QA  \
0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
1  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   

    QR   QN   QD   QC   QE   QQ   QG   QH   QI   QL   QK   QM   QF   QP   QS  \
0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
1  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   

    QT   QW   QY   QV   GA   GR   GN   GD   GC   GE   GQ     GG   GH    GI  \
0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  11.11  0.0  0.00   
1  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   0.00  0.0  6.67   

    GL   GK   GM   GF   GP   GS   GT    GW    GY   GV   HA   HR   HN   HD  \
0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  5.56  5.56  0.0  0.0  0.0  0.0  0.0   
1  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.00  0.00  0.0  0.0  0.0  0.0  0.0   

    HC   HE   HQ   HG   HH    HI   HL   HK   HM   HF   HP   HS   HT   HW   HY  \
0  0.0  0.0  0.0  0.0  0.0  0.00  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
1  0.0  0.0  0.0  0.0  0.0  6.67  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   

    HV    IA   IR   IN   ID   IC   IE 

In [93]:
# Verificación de campos
dataN.head(2)

Unnamed: 0 sequence  length  molecular_weight  charge  charge_density  \
0           0  MAATRNK       7            790.93   1.996        0.002524   
1           1  MKMDGLR       7            850.06   0.996        0.001172   

   isoelectric_point     gravy  instability_index  aromaticity  \
0          11.000305 -1.014286          77.657143          0.0   
1           8.497986 -0.671429          82.657143          0.0   

   aliphatic_index  boman_index  hydrophobic_ratio       A       R       N  \
0        28.571429     3.387143           0.428571  28.571  14.286  14.286   
1        55.714286     2.661429           0.428571   0.000  14.286   0.000   

        D    C    E    Q       G    H    I       L       K       M    F    P  \
0   0.000  0.0  0.0  0.0   0.000  0.0  0.0   0.000  14.286  14.286  0.0  0.0   
1  14.286  0.0  0.0  0.0  14.286  0.0  0.0  14.286  14.286  28.571  0.0  0.0   

     S       T    W    Y    V     AA   AR   AN   AD   AC   AE   AQ   AG   AH  \
0  0.0  14.286  0.0  0.0  0.0  16.67  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
1  0.0   0.000  0.0  0.0  0.0   0.00  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   

    AI   AL   AK   AM   AF   AP   AS     AT   AW   AY   AV   RA   RR     RN  \
0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  16.67  0.0  0.0  0.0  0.0  0.0  16.67   
1  0.0  0.0  0.0  0.0  0.0  0.0  0.0   0.00  0.0  0.0  0.0  0.0  0.0   0.00   

    RD   RC   RE   RQ   RG   RH   RI   RL   RK   RM   RF   RP   RS   RT   RW  \
0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
1  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   

    RY   RV   NA   NR   NN   ND   NC   NE   NQ   NG   NH   NI   NL     NK  \
0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  16.67   
1  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   0.00   

    NM   NF   NP   NS   NT   NW   NY   NV   DA   DR   DN   DD   DC   DE   DQ  \
0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
1  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   

      DG   DH   DI   DL   DK   DM   DF   DP   DS   DT   DW   DY   DV   CA  \
0   0.00  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
1  16.67  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   

    CR   CN   CD   CC   CE   CQ   CG   CH   CI   CL   CK   CM   CF   CP   CS  \
0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
1  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   

    CT   CW   CY   CV   EA   ER   EN   ED   EC   EE   EQ   EG   EH   EI   EL  \
0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
1  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   

    EK   EM   EF   EP   ES   ET   EW   EY   EV   QA   QR   QN   QD   QC   QE  \
0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
1  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   

    QQ   QG   QH   QI   QL   QK   QM   QF   QP   QS   QT   QW   QY   QV   GA  \
0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
1  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   

    GR   GN   GD   GC   GE   GQ   GG   GH   GI     GL   GK   GM   GF   GP  \
0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   0.00  0.0  0.0  0.0  0.0   
1  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  16.67  0.0  0.0  0.0  0.0   

    GS   GT   GW   GY   GV   HA   HR   HN   HD   HC   HE   HQ   HG   HH   HI  \
0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
1  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   

    HL   HK   HM   HF   HP   HS   HT   HW   HY   HV   IA   IR   IN   ID   IC  \
0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
1  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0 

No se logra ver en las 1761 columnas la variable respuesta, entonces se adicionará la columna Respuesta=0 en el caso de los negativos y Respuesta=1 en el caso de los positivos.  Luego se unirán los datos en una nuevo Dataframe unificado

In [94]:
dataN['Respuesta']=0

In [95]:
dataP['Respuesta']=1

In [96]:
data = dataN.append(dataP)

In [97]:
len_id=data['sequence'].unique

In [98]:
len_id

<bound method Series.unique of 0                        MAATRNK
1                        MKMDGLR
2                        MNIKHYQ
3                        LLWNSIP
4                        MHPHMSY
                  ...           
8317             ILPIIGKILSTIFGK
8318                   YAGAVANDL
8319             KKCFRWQWRMKKLGA
8320          GWAGWLLSPRGSRPSWGP
8321    LRRLYKRVVKLWKRLFRQLRRPVR
Name: sequence, Length: 13883, dtype: object>

la variable sequence es el idenficicador unico de los datos

In [99]:
#Se extrae una lista de las variables que contengan valores nulos
vars_with_na = [var for var in data.columns if data[var].isnull().sum()>1]
vars_with_na

[]

Significa que no hay variables con valores nulos

## Análisis Descriptivo de los Datos

In [100]:
data.shape

(13883, 1762)

In [101]:
# lista de varaibles numericas
num_vars = [var for var in data.columns if data[var].dtypes != 'O']  #'O' significa python objects
len(num_vars)

1761

Todas las variables son numéricas excepto 1, que es la sequencia, que actua como identificador de la molécula

## Preprocesamiento

In [102]:
data.shape

(13883, 1762)

In [104]:
# Separación de los datos en train y test
# random_state para reproducibilidad

X_train, X_test, y_train, y_test = train_test_split(data, data.Respuesta,
                                                    test_size=0.2,
                                                    random_state=0)
X_train.shape, X_test.shape

((11106, 1762), (2777, 1762))

In [105]:
#varaibles numericas a estandarizar
train_vars = [var for var in X_train.columns if var not in ['sequence']]
len(train_vars)

1761

In [106]:
#ajuste del escalador
scaler = StandardScaler()  #se crea una instancia del escalador
scaler.fit(X_train[train_vars]) #  se ajusta el escalador a los datos de entrenamiento

# Se transforman los datos de train y test con el escalador ajustado a los datos de entrenamiento
X_train = pd.concat([X_train[['sequence']].reset_index(drop=True),
                    pd.DataFrame(scaler.transform(X_train[num_vars]), columns=train_vars)],
                    axis=1)

X_test = pd.concat([X_test[['sequence']].reset_index(drop=True),
                    pd.DataFrame(scaler.transform(X_test[num_vars]), columns=train_vars)],
                    axis=1)


In [111]:
# Se guardan los datos estandarizados de test y train

X_train.to_pickle('./Datos/xtrain.pkl')
X_test.to_pickle('./Datos/xtest.pkl')
y_train.to_pickle('./Datos/ytrain.pkl')
y_test.to_pickle('./Datos/ytest.pkl')


## Selección y Reducción de Variables

In [126]:
#se cargan los datos estandarizados
X_train = pd.read_pickle('./Datos/xtrain.pkl')
X_test = pd.read_pickle('./Datos/xtest.pkl')
y_train = pd.read_pickle('./Datos/ytrain.pkl')
y_test = pd.read_pickle('./Datos/ytest.pkl')

Se usará una regresión lasso,, con una semilla = 0, y la función selectFromModel de sklearn para seleccionar las varaibles más predictivas.  Entre mayor alpha más reducción de variables

In [127]:
#lr = LogisticRegression(solver='liblinear')

In [128]:
X_train.head(2)

sequence  Unnamed: 0    length  molecular_weight  \
0  VDIGGYYRPDPLKTAAAMRPSATLNAIIDAL    0.014044  1.672052          1.388966   
1           FFWHHIGHALDAAKRVHGMLSG    1.332868  0.454519          0.424816   

     charge  charge_density  isoelectric_point     gravy  instability_index  \
0 -0.695765       -0.711994          -1.012555  0.274114          -0.546592   
1 -0.226782       -0.310505           0.044100  0.198001           0.247031   

   aromaticity  aliphatic_index  boman_index  hydrophobic_ratio         A  \
0    -0.464725         0.309426    -0.242920           0.243913  1.469908   
1     0.235414        -0.191699    -0.411259           0.263569  0.780374   

          R         N         D         C         E         Q         G  \
0 -0.155832 -0.004870  1.406158 -0.517341 -0.580968 -0.576528 -0.041592   
1 -0.340374 -0.681305  0.356929 -0.517341 -0.580968 -0.576528  0.940965   

          H         I         L         K         M         F         P  \
0 -0.466195  0.561009 -0.110090 -0.587903  0.035070 -0.774340  0.784709   
1  3.172222 -0.207692 -0.167901 -0.470077  0.344896  0.577482 -0.661013   

          S         T         W         Y         V        AA        AR  \
0 -0.340692  0.416038 -0.445862  0.877191 -0.389453  0.927881 -0.227468   
1 -0.132514 -0.749556  0.148386 -0.593239 -0.191101  1.455210 -0.227468   

         AN        AD        AC        AE        AQ        AG       AH  \
0 -0.169437 -0.169749 -0.150238 -0.176008 -0.152632 -0.292895 -0.14596   
1 -0.169437 -0.169749 -0.150238 -0.176008 -0.152632 -0.292895 -0.14596   

         AI        AL        AK        AM        AF        AP        AS  \
0  1.689230  0.887227 -0.280909  3.823798 -0.206587 -0.182117 -0.255379   
1 -0.255096  1.412617  1.522537 -0.120794 -0.206587 -0.182117 -0.255379   

         AT        AW        AY        AV        RA        RR        RN  \
0  2.026351 -0.138193 -0.143163 -0.261894 -0.207675 -0.318107 -0.196897   
1 -0.208479 -0.138193 -0.143163 -0.261894 -0.207675 -0.318107 -0.196897   

         RD        RC        RE        RQ        RG        RH        RI  \
0 -0.166939 -0.199194 -0.161435 -0.163783 -0.250665 -0.134579 -0.240462   
1 -0.166939 -0.199194 -0.161435 -0.163783 -0.250665 -0.134579 -0.240462   

         RL        RK        RM        RF        RP        RS        RT  \
0 -0.277647 -0.269359 -0.114753 -0.224142  3.324259 -0.203576 -0.192207   
1 -0.277647 -0.269359 -0.114753 -0.224142 -0.177877 -0.203576 -0.192207   

         RW        RY        RV        NA        NR        NN       ND  \
0 -0.197519 -0.156772 -0.222273  2.437105 -0.182614 -0.146734 -0.12537   
1 -0.197519 -0.156772  2.281897 -0.190719 -0.182614 -0.146734 -0.12537   

         NC        NE        NQ        NG        NH        NI        NL  \
0 -0.121745 -0.127618 -0.130867 -0.172984 -0.118378 -0.174682 -0.220796   
1 -0.121745 -0.127618 -0.130867 -0.172984 -0.118378 -0.174682 -0.220796   

         NK       NM        NF        NP        NS        NT       NW  \
0 -0.185218 -0.10318 -0.159625 -0.142695 -0.175774 -0.155238 -0.12157   
1 -0.185218 -0.10318 -0.159625 -0.142695 -0.175774 -0.155238 -0.12157   

         NY        NV        DA        DR        DN        DD        DC  \
0 -0.125746 -0.171583  2.780511 -0.145453 -0.122463 -0.110409 -0.091754   
1 -0.125746 -0.171583  4.050298 -0.145453 -0.122463 -0.110409 -0.091754   

         DE        DQ        DG        DH        DI        DL        DK  \
0 -0.135743 -0.102083 -0.156331 -0.087176  3.024230 -0.201215 -0.175506   
1 -0.135743 -0.102083 -0.156331 -0.087176 -0.165977 -0.201215 -0.175506   

         DM        DF        DP        DS        DT        DW        DY  \
0 -0.090122 -0.155797  2.757323 -0.151051 -0.158733 -0.095897 -0.116966   
1 -0.090122 -0.155797 -0.138504 -0.151051 -0.158733 -0.095897 -0.116966   

         DV        CA        CR        CN        CD        CC        CE  \
0 -0.169904 -0.158238 -0.188358 -0.133494 -0.109646 -0.112791 -0.102987   
1 -0.169904 -0.1582

In [129]:
X_train.drop(['sequence', 'Respuesta'], axis=1, inplace=True)

In [130]:
sel_ = SelectFromModel(Lasso(alpha=0.05, random_state=0)) # remember to set the seed, the random state in this function
sel_.fit(X_train, y_train)

SelectFromModel(estimator=Lasso(alpha=0.05, copy_X=True, fit_intercept=True,
                                max_iter=1000, normalize=False, positive=False,
                                precompute=False, random_state=0,
                                selection='cyclic', tol=0.0001,
                                warm_start=False),
                max_features=None, norm_order=1, prefit=False, threshold=None)

In [131]:
sel_.get_support()  #para visualizar las varaibles seleccionas

array([ True, False, False, ..., False, False, False])

In [132]:
sum(sel_.get_support())  #se seleccionaron 15 variables significativas según el modelo RL Lasso

15

In [133]:
selected_feat = X_train.columns[(sel_.get_support())]

In [134]:
selected_feat  #caracteristicas seleccionadas

Index(['Unnamed: 0', 'charge_density', 'C', 'K', 'M',
       '_SolventAccessibilityC1', '_SolventAccessibilityC3',
       '_SolventAccessibilityD1001', '_SolventAccessibilityD3001',
       '_SecondaryStrD1001', '_ChargeD3100', 'MoranAuto_Hydrophobicity2',
       'QSOgrant27', 'embed_0_51', 'embed_1_57'],
      dtype='object')

In [137]:
# algunas estadisticas
print('total variables: {}'.format((X_train.shape[1])))
print('varaibles seleccionadas: {}'.format(len(selected_feat)))
print('Variables excluidas por RL Lasso: {}'.format(
    np.sum(sel_.estimator_.coef_ == 0)))

total variables: 1760
varaibles seleccionadas: 15
Variables excluidas por RL Lasso: 1745


In [140]:
selected_feats = X_train.columns[(sel_.estimator_.coef_ != 0).ravel().tolist()]
selected_feats  #otra forma de ver las caracteriticas seleccionadas

Index(['Unnamed: 0', 'charge_density', 'C', 'K', 'M',
       '_SolventAccessibilityC1', '_SolventAccessibilityC3',
       '_SolventAccessibilityD1001', '_SolventAccessibilityD3001',
       '_SecondaryStrD1001', '_ChargeD3100', 'MoranAuto_Hydrophobicity2',
       'QSOgrant27', 'embed_0_51', 'embed_1_57'],
      dtype='object')

In [141]:
# nse guarda la lista de las caracteristicas seleccionadas
pd.Series(selected_feats).to_csv('./Datos/selected_features.csv', index=False)

C:\Users\Usuario\Anaconda3\envs\patrones\lib\site-packages\ipykernel_launcher.py:2: FutureWarning: The signature of `Series.to_csv` was aligned to that of `DataFrame.to_csv`, and argument 'header' will change its default value from False to True: please pass an explicit value to suppress this warning.
  


## Modelo 1 - Regresión Logística

## Modelo 2 - Arboles de Clasificación

## Modelo 3 - Máquina de Vectores de Soporte

## Modelo 4 - Adaboost